pep/app/services/current-user.ts

```diff
@@ -54,12 +54,10 @@ export default class CurrentUserService extends Service {
      * @returns Promise<User | void>
      */
     async fetchUser(): Promise<User | void> {
-        if (this.session.isAuthenticated) {
```

Comment:
we get a User back from this endpoint even if you send in an unauthed SessionId?

And if thats the case, wouldn't you need to use the session id from `getUnauthenticatedSession()` if they dont have an authenticated session id?

Reply:
related to that: I believe we need to update anywhere we are using `session.data.authenticated.SessionId` to fallback to using `getUnauthenticatedSession().SessionId` if the authenticated session ID is empty (e.g. in the ajax service, application adapter). 

Thats how the API will be able to use the session id to track the actions of unauthenticated/anonymous users.

Reply:
No we don't - but this method is only requested when your logged in (in the application route)

Reply:
Gotcha. Though, I think its still better to have that guard directly in the method too, because it shouldn't attempt the request no matter where its triggered from, if we wanted to call it from somewhere else in the app.